src/AzureMcp.csproj

```diff
@@ -42,33 +39,58 @@
-    <PackageReference Include="ModelContextProtocol.AspNetCore" Version="0.1.0-preview.9" />
-    <PackageReference Include="Microsoft.Extensions.Caching.Memory" Version="9.0.4" />
-    <PackageReference Include="Microsoft.Azure.Cosmos" Version="3.48.0" />
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="9.0.4" />
-    <PackageReference Include="Microsoft.Extensions.Hosting" Version="9.0.4" />
-    <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
-    <PackageReference Include="System.CommandLine" Version="2.0.0-beta4.22272.1" />
-    <PackageReference Include="System.Formats.Asn1" Version="9.0.4" />
-    <PackageReference Include="System.Linq.AsyncEnumerable" Version="10.0.0-preview.3.25171.5" />
-    <PackageReference Include="System.Net.ServerSentEvents" Version="10.0.0-preview.3.25171.5" />
-    <PackageReference Include="OpenTelemetry.Exporter.OpenTelemetryProtocol" Version="1.11.2" />
-    <PackageReference Include="Azure.Monitor.OpenTelemetry.AspNetCore" Version="1.2.0" />
-    <PackageReference Include="Microsoft.Extensions.Azure" Version="1.11.0" />
+    <PackageReference Include="Azure.Core" />
+    <PackageReference Include="Azure.Data.AppConfiguration" />
+    <PackageReference Include="Azure.Identity" />
+    <PackageReference Include="Azure.Data.Tables" />
+    <PackageReference Include="Azure.Identity.Broker" />
+    <PackageReference Include="Azure.Monitor.Query" />
+    <PackageReference Include="Azure.ResourceManager.AppConfiguration" />
+    <PackageReference Include="Azure.Security.KeyVault.Keys" />
+    <PackageReference Include="Azure.Storage.Blobs" />
+    <PackageReference Include="Azure.ResourceManager.CosmosDB" />
+    <PackageReference Include="Azure.ResourceManager.OperationalInsights" />
+    <PackageReference Include="Azure.ResourceManager.Storage" />
+    <PackageReference Include="Microsoft.Identity.Client.Broker" />
+    <PackageReference Include="ModelContextProtocol" />
+    <PackageReference Include="ModelContextProtocol.AspNetCore" />
+    <PackageReference Include="Microsoft.Extensions.Caching.Memory" />
+    <PackageReference Include="Microsoft.Azure.Cosmos" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" />
+    <PackageReference Include="Microsoft.Extensions.Hosting" />
+    <PackageReference Include="Newtonsoft.Json" />
+    <PackageReference Include="System.CommandLine" />
+    <PackageReference Include="System.Formats.Asn1" />
+    <PackageReference Include="System.Linq.AsyncEnumerable" />
+    <PackageReference Include="System.Net.ServerSentEvents" />
+    <PackageReference Include="OpenTelemetry.Exporter.OpenTelemetryProtocol" />
+    <PackageReference Include="Azure.Monitor.OpenTelemetry.AspNetCore" />
+    <PackageReference Include="Microsoft.Extensions.Azure" />
+  </ItemGroup>
+
+  <!-- Runs before every dotnet build -->
+  <Target Name="InstallGitHooks" BeforeTargets="BeforeBuild">
```

Comment:
I like the idea of putting the precommit hook installer in a place where we know it'll get invoked, but I always hate to see the loss of a vanilla .csproj file.  Msbuild always comes with a level of indirection that borders on magic.

I'd prefer we keep things unrelated to building binaries out of msbuild and in the powershell scripts. i.e.  proj files for making the dll, scripts for all other orchestration.  But, I don't know of a way to ensure that visual-studio-only devs also get the hooks installed.

Lets merge this as is, then see if we can move the target out of csproj and into the /eng folder.